packages/babel-plugin-transform-parameters/src/rest.js

```diff
@@ -153,6 +153,17 @@ const memberExpressionOptimisationVisitor = {
     }
   },
 };
+
+function getParamsCount(node) {
+  let count = node.params.length;
+  // skip the first parameter if it is a TypeScript 'this parameter'
```

Comment:
Can you guard the following with `if (count)` ?, accessing `0` of empty array will result in expensive prototype chain lookup. https://v8.dev/blog/elements-kinds#avoid-reading-beyond-the-length-of-the-array